modules/user_accounts/php/edit_user.class.inc

```diff
@@ -1045,10 +1045,10 @@ class Edit_User extends \NDB_Form
             && (!empty($values['Password_hash']) || !empty($values['__Confirm']))
         ) {
             try {
-                $password = new \Password($values['Password_hash']);
                 // New password must be different than current one
+                $password        = new \Password($values['Password_hash']);
                 $passwordChanged = \User::factory($this->identifier)
-                    ->passwordChanged($_POST['password']);
+                    ->passwordChanged($values['Password_hash']);
```

Comment:
Whether or not this fixes the bug, removing a superglobal reference is always a step forward..